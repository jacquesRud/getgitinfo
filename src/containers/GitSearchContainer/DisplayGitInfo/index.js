import { useMemo } from "react";

export default function DisplayGitInfo({ data }) {
  const display = useMemo(() => {
    return !data.total_count
      ? "No User Found"
      : data.items.map(({ login }) => <div>{login}</div>);
  }, [data]);
  return display;
}
