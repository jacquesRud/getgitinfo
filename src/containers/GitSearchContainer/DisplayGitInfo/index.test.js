import { render } from "@testing-library/react";
import DisplayGitInfo from "./";

describe("DisplayGitInfos", () => {
  test("Render Empty", () => {
    const { container } = render(<DisplayGitInfo data={[]} />);
    expect(container).toMatchSnapshot();
  });
  test("Render", () => {
    const { container } = render(
      <DisplayGitInfo data={{ items: ["a", ",b", "c"] }} />
    );
    expect(container).toMatchSnapshot();
  });
});
