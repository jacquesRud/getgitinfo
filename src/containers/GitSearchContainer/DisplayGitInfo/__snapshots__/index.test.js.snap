// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayGitInfos Render 1`] = `
<div>
  No User Found
</div>
`;

exports[`DisplayGitInfos Render Empty 1`] = `
<div>
  No User Found
</div>
`;
